Framework/Core/src/DataProcessingDevice.cxx

```diff
@@ -991,23 +991,35 @@ void DataProcessingDevice::fillContext(DataProcessorContext& context, DeviceCont
   /// We must make sure there is no optional
   /// if we want to optimize the forwarding
   context.canForwardEarly = (spec.forwards.empty() == false) && mProcessingPolicies.earlyForward != EarlyForwardPolicy::NEVER;
+  bool onlyConditions = false;
+  bool overriddenEarlyForward = false;
   for (auto& forwarded : spec.forwards) {
+    if (forwarded.matcher.lifetime == Lifetime::Condition) {
+      onlyConditions = true;
```

Comment:
isn't this logic for detecting onlyConditions wrong, since it seems inverted?

Reply:
you are right, fixed.